src/ui/mappixmapitem.cpp

```diff
@@ -89,20 +90,12 @@ void MapPixmapItem::shift(int xDelta, int yDelta, bool fromScriptCall) {
         destY %= map->getHeight();
 
         int blockIndex = j * map->getWidth() + i;
-        Block srcBlock = backupBlockdata->blocks->at(blockIndex);
+        Block srcBlock = oldMetatiles.at(blockIndex);
```

Comment:
I think the logic is incorrect here.  If `fromScriptCall` is `true`, then this will segfault due to out-of-bounds access.

Reply:
Also, I just noticed that the caller of this function isn't passing `fromScriptCall = true`.  Could you tack that onto this PR?  It's in `mainwindow_scriptapi.cpp`.

Reply:
Yeah you're right. I assumed this was following the same structure as the other functions that copy the old metatiles, but didn't think about the fact that this one accesses them. And yes I'll make that change in `mainwindow_scriptapi.cpp`.